src/input.rs

```diff
@@ -25,29 +25,38 @@ impl GraphConverter {
     /// Takes in the parameters for the new input, construct the node in RexExpr,
     /// return the Id (index) of this input node in the RecExpr. This is the
     /// pattern for all these op functions.
-    pub fn new_input(&mut self, name: &str, dim1: i32, dim2: i32, dim3: i32, dim4: i32) -> Id {
+    pub fn new_input(&mut self, dims: Vec<i32>) -> Id {
```

Comment:
Also, if you use a slice, you can change all uses of `new_input` to avoid creating a vector when the size of the vector is statically known:
```
new_input(vec![foo, bar]) ==> new_input(&[foo, bar])
```
This is more efficient as it allocates on the stack rather that the heap.

Reply:
Thanks! Changed this